TESTS/events/equeue/main.cpp

```diff
@@ -0,0 +1,1023 @@
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+
+using namespace utest::v1;
+
+#define TEST_EQUEUE_SIZE (4*EVENTS_EVENT_SIZE)
+#define TEST_THREAD_STACK_SIZE 512
+
+// Test functions
+static void pass_func(void *eh)
+{
+}
+
+static void simple_func(void *p)
+{
+    (*(reinterpret_cast<uint8_t *>(p)))++;
+}
+
+static void sloth_func(void *p)
+{
+    ThisThread::sleep_for(10);
+    (*(reinterpret_cast<uint8_t *>(p)))++;
+}
+
+struct indirect {
+    uint8_t *touched;
+    uint8_t buffer[7];
+};
+
+static void indirect_func(void *p)
+{
+    struct indirect *i = reinterpret_cast<struct indirect *>(p);
+    (*i->touched)++;
+}
+
+struct timing {
+    unsigned tick;
+    unsigned delay;
+};
+
+static void timing_func(void *p)
+{
+    struct timing *timing = reinterpret_cast<struct timing *>(p);
+    unsigned tick = equeue_tick();
+
+    unsigned t1 = timing->delay;
+    unsigned t2 = tick - timing->tick;
+    TEST_ASSERT_UINT_WITHIN(10, t2, t1);
+
+    timing->tick = tick;
+}
+
+struct fragment {
+    equeue_t *q;
+    size_t size;
+    struct timing timing;
+};
+
+static void fragment_func(void *p)
+{
+    struct fragment *fragment = reinterpret_cast<struct fragment *>(p);
+    timing_func(&fragment->timing);
+
+    struct fragment *nfragment = reinterpret_cast<struct fragment *>(equeue_alloc(fragment->q, fragment->size));
+    TEST_ASSERT_NOT_NULL(nfragment);
+
+    *nfragment = *fragment;
+    equeue_event_delay(nfragment, fragment->timing.delay);
+
+    int id = equeue_post(nfragment->q, fragment_func, nfragment);
+    TEST_ASSERT_NOT_EQUAL(0, id);
+}
+
+struct cancel {
+    equeue_t *q;
+    int id;
+};
+
+static void cancel_func(void *p)
+{
+    struct cancel *ccel = reinterpret_cast<struct cancel *>(p);
+    equeue_cancel(ccel->q, ccel->id);
+}
+
+struct nest {
+    equeue_t *q;
+    void (*cb)(void *);
+    void *data;
+};
+
+static void nest_func(void *p)
+{
+    struct nest *nst = reinterpret_cast<struct nest *>(p);
+    equeue_call(nst->q, nst->cb, nst->data);
+
+    ThisThread::sleep_for(10);
+}
+
+static void multithread_thread(equeue_t *p)
+{
+    equeue_dispatch(p, -1);
+}
+
+static void background_func(void *p, int ms)
+{
+    *(reinterpret_cast<int *>(p)) = ms;
+}
+
+struct ethread {
+    equeue_t *q;
+    int ms;
+};
+
+static void ethread_dispatch(void *p)
+{
+    struct ethread *t = reinterpret_cast<struct ethread *>(p);
+    equeue_dispatch(t->q, t->ms);
+}
+
+struct count_and_queue {
+    int p;
+    equeue_t *q;
+};
+
+static void simple_breaker(void *p)
+{
+    struct count_and_queue *caq = reinterpret_cast<struct count_and_queue *>(p);
+    equeue_break(caq->q);
+    ThisThread::sleep_for(10);
+    caq->p++;
+}
+
+// Simple call tests
+
+/** Test that equeue executes function passed by equeue_call.
+ * 
+ *  Given queue is initialized.
+ *  When the event is scheduled and after that equeue_dispatch is called.
+ *  Then function passed by equeue_call is executed properly.
+ */
+static void simple_call_test()
```

Comment:
Change test functions naming `simple_call_test` to `test_equeue_simple_call`
Apply to all test cases